File: app/src/main/java/com/aero51/moviedatabase/repository/model/credits/MovieCredits.java
package com.aero51.moviedatabase.repository.model.credits;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.List;

@Entity(tableName = "movie_credit")
public class MovieCredits {
    @PrimaryKey(autoGenerate = false)
    private Integer id;

    @Ignore
    private List<Cast> cast;

    @Ignore
    private List<Crew> crew;

    @Ignore
    public MovieCredits(Integer id, List<Cast> cast, List<Crew> crew) {
        this.id = id;
        this.cast = cast;
        this.crew = crew;
    }

    public MovieCredits(Integer id) {
        this.id = id;
    }

    public Integer getId() {
        return id;
    }

    public List<Cast> getCast() {
        return cast;
    }

    public List<Crew> getCrew() {
        return crew;
    }

    public void setCast(List<Cast> cast) { this.cast = cast; }

    public void setCrew(List<Crew> crew) { this.crew = crew; }
}

File: app/src/main/java/com/aero51/moviedatabase/repository/model/tmdb/credits/MovieCredits.kt
package com.aero51.moviedatabase.repository.model.tmdb.credits

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import java.io.Serializable

@Entity(tableName = "movie_credits")
class MovieCredits {
    @PrimaryKey(autoGenerate = false)
    var id: Int? = null

    @Ignore
    @SerializedName("cast")
    var movieCast: List<MovieCast>? = null

    @Ignore
    @SerializedName("crew")
    var movieCrew: List<MovieCrew>? = null

    @Entity(tableName = "movie_crew")
    class MovieCrew {
        @PrimaryKey(autoGenerate = true)
        var db_id: Int? = null
        var credit_id: String? = null
        var department: String? = null
        var gender: Int? = null
        var id: Int? = null
        var job: String? = null
        var name: String? = null
        var profile_path: String? = null
        var movie_id: Int? = null
    }

    @Entity(tableName = "movie_cast")
    class MovieCast : Serializable {
        @PrimaryKey(autoGenerate = true)
        var db_id: Int? = null
        var cast_id: Int? = null
        var character: String? = null
        var credit_id: String? = null
        var gender: Int? = null
        var id: Int? = null
        var name: String? = null
        var order: Int? = null
        var profile_path: String? = null
        var movie_id: Int? = null
    }
}